File: src/main/java/com/chat/messages/domain/repository/MessageRepository.java
package com.chat.messages.domain.repository;

import com.chat.messages.domain.model.Message;
import java.util.List;
import java.util.Optional;

/** Persistence port for {@link com.chat.messages.domain.model.Message} aggregates. */
public interface MessageRepository {

  Message save(Message message);

  Optional<Message> findById(String id);

  List<Message> findByChatId(String chatId, int offset, int limit);

  long countByChatId(String chatId);

  long countAll();
}

File: services/server/src/messages/domain/repository/message.repository.ts
import { Message } from "../model/message";

export interface MessageRepository {
  save(message: Message): Promise<Message>;
  findById(id: string): Promise<Message | null>;
  findByChatId(
    chatId: string,
    options: { page: number; limit: number; search?: string },
  ): Promise<Message[]>;
}
